app/sign-out/controllers.js

```diff
@@ -0,0 +1,21 @@
+const { IS_PRODUCTION, SERVER_HOST, AUTH } = require('../../config')
+
+function buildRedirectUrl () {
+  const protocol = IS_PRODUCTION ? 'https' : 'http'
+  return `${protocol}://${SERVER_HOST}`
+}
+
+function buildLogoutUrl () {
+  const postLogoutRedirectUrl = buildRedirectUrl()
+  const path = `/auth/logout?client_id=${AUTH.PROVIDER_KEY}&redirect_uri=${postLogoutRedirectUrl}`
+  return new URL(path, AUTH.PROVIDER_URL).href
```

Comment:
I’ve found this can blow up if this env isn’t set, so good to protect with a ternary like this:

```js
return AUTH.PROVIDER_URL ? new URL(path, AUTH.PROVIDER_URL).href : ‘’
```

Reply:
Not #115 is also in I’d add the logout path to the hmpps object in the config.

Then you can append the redirect in the controller, where you’ll also have access to the full URL of the app I think so you won’t have to use the `SERVER_HOST` variable.

Reply:
👍